cartridges/int_bolt_sfra_ppc/cartridge/client/default/js/boltProductPageButton.js

```diff
@@ -0,0 +1,295 @@
+                    value: sfccData.sfcc.sfcc_order_token
+                });
+
+            redirect.submit();
+        }
+    },
+    onCheckoutStart: function () {
+        // This function is called after the checkout form is presented to the user.
+    },
+
+    // eslint-disable-next-line no-unused-vars
+    onEmailEnter: function (email) {
+        // This function is called after the user enters their email address.
+    },
+
+    onShippingDetailsComplete: function () {
+        // This function is called when the user proceeds to the shipping options page.
+        // This is applicable only to multi-step checkout.
+    },
+
+    onShippingOptionsComplete: function () {
+        // This function is called when the user proceeds to the payment details page.
+        // This is applicable only to multi-step checkout.
+    },
+
+    onPaymentSubmit: function () {
+        // This function is called after the user clicks the pay button.
+    },
+    success: function (transaction, callback) {
+        // This function is called when the Bolt checkout transaction is successful.
+        sfccData = transaction;
+        console.log(`----------- ${JSON.stringify(transaction)} -----------`);
+        callback();
+    }
+};
+
+var showPPCButton = function () {
+    setupPPCButtonPoller = setInterval(function () {
+        var ppcBoltButton = $('.bolt-button-wrapper[data-tid="instant-bolt-checkout-button"]'); // @ts-ignore
+        var ppcButtonLoaded = ppcBoltButton && window.BoltCheckout && ppcBoltButton.children()[0].nodeName === 'svg';
+        if (ppcButtonLoaded) {
+            ppcBoltButton[0].style.display = '';
+            ppcButtonVisible = true;
+            clearInterval(setupPPCButtonPoller);
+        }
+    }, 250);
+};
+
+var hidePPCButton = function () {
+    if (setupPPCButtonPoller) {
+        clearInterval(setupPPCButtonPoller);
+    }
+    var ppcBoltButton = $('.bolt-button-wrapper[data-tid="instant-bolt-checkout-button"]'); // @ts-ignore
+    if (ppcBoltButton) {
+        ppcBoltButton[0].style.display = 'None';
+    }
+    ppcButtonVisible = false;
+};
+
+// configurePPCCart should be called as soon as the product is ready to be ordered and
+// whenever the product's attributes change.
+var configurePPCCartAndShowButton = function (boltCartObject) {
+    if (!window.BoltCheckout) {
+        console.error('Unable to call BoltConnect.ConfigureProductCheckout since window.BoltCheckout is defined');
+        return;
+    }
+    ppcCart = boltCartObject;
+    var result = window.BoltCheckout.configureProductCheckout(
+        boltCartObject,
+        {}, // Hints
+        callbacks,
+        { checkoutButtonClassName: ppcButtonClass }
+    );
+    if (result instanceof Error) {
+        console.error(`Call to BoltCheckout.configureProductCheckout returned an error ${result.toString()}`);
+        return;
+    }
+    console.log('Call to BoltCheckout.configureProductCheckout was successful');
```

Comment:
Is keeping this log intentional?

Reply:
Yes, this is fine as it doesn't expose sensitive data.